### ISSUE TITLE
Error adding container to network "macvlan-conf": address already in use

### ISSUE THREAD

**Issue by mr-yaky** [k8snetworkplumbingwg/multus-cni#861]
**What happend**:
We used private IPs with type `host-local` and did't get any problems.
Now we need to use multus with `macvlan` and type `static` and with custom MAC/IP address for the second pod interface. We want to a get public ip inside a pod but we get the next problem: 

```
Events:
  Type     Reason                  Age                    From     Message
  ----     ------                  ----                   ----     -------
  Normal   AddedInterface          22m                    multus   Add eth0 [10.233.113.129/32] from cni0
  Warning  FailedCreatePodSandBox  22m                    kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "b079fe27bb70bc8260f61476ffa55cf3e51edf03d3b650e732b9a5cdcf0cd05c" network for pod "samplepod": networkPlugin cni failed to set up pod "samplepod_playground" network: [playground/samplepod/:macvlan-conf]: error adding container to network "macvlan-conf": address already in use
  Normal   AddedInterface          22m                    multus   Add eth0 [10.233.113.183/32] from cni0
  Warning  FailedCreatePodSandBox  22m                    kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "61d2ca9356b53e439ec46a161d651cfcc994e6ed99c744667f9870033bce6994" network for pod "samplepod": networkPlugin cni failed to set up pod "samplepod_playground" network: [playground/samplepod/:macvlan-conf]: error adding container to network "macvlan-conf": address already in use
  Normal   AddedInterface          22m                    multus   Add eth0 [10.233.113.103/32] from cni0
  Warning  FailedCreatePodSandBox  22m                    kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "5c82877baba1571bf693e08f1dc3acd448c232f4b128119e469498c82ac0bbeb" network for pod "samplepod": networkPlugin cni failed to set up pod "samplepod_playground" network: [playground/samplepod/:macvlan-conf]: error adding container to network "macvlan-conf": address already in use
...
```
```
% kubectl -n playground get po samplepod
NAME        READY   STATUS              RESTARTS   AGE
samplepod   0/1     ContainerCreating   0          24m
```

**What you expected to happen**:
When I have a free public IP address I should be able to use it with a multus second network interface.

**Anything else we need to know?**:
1. K8s node is a VM running with KVM. We have the following network configuration for VM:
em3 (HW) --> br0 --> vnet8--> eth0 (VM node-04) -- this eth0 is using as master iface for macvlan
```
% kubectl get node node-04 -o wide
NAME       STATUS   ROLES    AGE      VERSION   INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                KERNEL-VERSION                CONTAINER-RUNTIME
node-04   Ready    <none>   2y334d   v1.22.8   1.1.1.249   <none>        CentOS Linux 7 (Core)   3.10.0-1160.53.1.el7.x86_64   docker://20.10.11
```
```
# lspci | grep -i network
00:07.0 Ethernet controller: Red Hat, Inc. Virtio network device
```
```
# modinfo virtio_net
filename:       /lib/modules/3.10.0-1160.53.1.el7.x86_64/kernel/drivers/net/virtio_net.ko.xz
license:        GPL
description:    Virtio network driver
retpoline:      Y
rhelversion:    7.9
srcversion:     3C2214E3B5BBD7B8DFC06D9
alias:          virtio:d00000001v*
depends:        virtio,net_failover,virtio_ring
intree:         Y
vermagic:       3.10.0-1160.53.1.el7.x86_64 SMP mod_unload modversions
signer:         CentOS Linux kernel signing key
sig_key:        87:E5:16:0E:55:99:B4:50:B2:48:34:5F:17:61:A1:76:E0:8A:6D:FD
sig_hashalgo:   sha256
parm:           napi_weight:int
parm:           csum:bool
parm:           gso:bool
```

2. When I'm trying to ping sometimes it provides me a response but only when I have `samplepod` as running.
Without `samplepod` this IP is free. Also I have tested it in the separated KVM vm in the same public bridge and it works well. 

```
% ping 1.1.1.247
PING 1.1.1.247 (1.1.1.247): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
Request timeout for icmp_seq 3
Request timeout for icmp_seq 4
64 bytes from 1.1.1.247: icmp_seq=5 ttl=53 time=43.482 ms
Request timeout for icmp_seq 6
Request timeout for icmp_seq 7
Request timeout for icmp_seq 8
Request timeout for icmp_seq 9
^C
--- 1.1.1.247 ping statistics ---
11 packets transmitted, 1 packets received, 90.9% packet loss
round-trip min/avg/max/stddev = 43.482/43.482/43.482/0.000 ms
```

3. I found that when I'm not setting MAC address on the pod annotations then the pod will be started without any issues. And when I put it back to the configuration it gives me an error. We have to use a specific MAC address with KVM to get our public IP as working.

**Environment**:

- Multus version: tested with v3.8, v3.8.1, latest docker images (same behaviour)
- Kubernetes version (use `kubectl version`):
```
...
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.8", GitCommit:"7061dbbf75f9f82e8ab21f9be7e8ffcaae8e0d44", GitTreeState:"clean", BuildDate:"2022-03-16T14:04:34Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
```
- Primary CNI for Kubernetes cluster:
Calico v3.20.3
- OS (e.g. from /etc/os-release):
CentOS Linux release 7.9.2009 (Core)
- File of '/etc/cni/net.d/'
```
{
  "cniVersion": "0.3.1",
  "name": "multus-cni-network",
  "type": "multus",
  "capabilities": {
    "portMappings": true,
    "bandwidth": true
  },
  "kubeconfig": "/etc/cni/net.d/multus.d/multus.kubeconfig",
  "delegates": [
    {
      "name": "cni0",
      "cniVersion": "0.3.1",
      "plugins": [
        {
          "nodename": "node-04",
          "type": "calico",
          "log_level": "info",
          "log_file_path": "/var/log/calico/cni/cni.log",
          "etcd_endpoints": "https://192.168.2.1:2379,https://192.168.2.2:2379,https://192.168.2.3:2379",
          "etcd_cert_file": "/etc/calico/certs/cert.crt",
          "etcd_key_file": "/etc/calico/certs/key.pem",
          "etcd_ca_cert_file": "/etc/calico/certs/ca_cert.crt",
          "ipam": {
            "type": "calico-ipam",
            "assign_ipv4": "true",
            "ipv4_pools": [
              "10.233.64.0/18"
            ]
          },
          "policy": {
            "type": "k8s"
          },
          "kubernetes": {
            "kubeconfig": "/etc/cni/net.d/calico-kubeconfig"
          }
        },
        {
          "type": "portmap",
          "capabilities": {
            "portMappings": true
          }
        },
        {
          "type": "bandwidth",
          "capabilities": {
            "bandwidth": true
          }
        }
      ]
    }
  ]
}
```
- File of '/etc/cni/multus/net.d'
--
- NetworkAttachment info (use `kubectl get net-attach-def -o yaml`)
```
apiVersion: v1
items: []
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
vitaly@gadol ansible % kubectl -n playground get net-attach-def -o yaml
apiVersion: v1
items:
- apiVersion: k8s.cni.cncf.io/v1
  kind: NetworkAttachmentDefinition
  metadata:
    creationTimestamp: "2022-06-10T17:23:58Z"
    generation: 1
    name: macvlan-conf
    namespace: playground
    resourceVersion: "906923352"
    uid: 1eb72219-ae7d-4abc-ba00-e139dad6cbcc
  spec:
    config: '{ "cniVersion": "0.3.1", "plugins": [ { "type": "macvlan", "capabilities":
      { "ips": true }, "master": "eth0", "mode": "bridge", "ipam": { "type": "static",
      "routes": [ { "dst": "0.0.0.0/0", "gw": "1.1.1.129" } ] } }, { "capabilities":
      { "mac": true }, "type": "tuning" } ] }'
```
- Target pod yaml info (with annotation, use `kubectl get pod <podname> -o yaml`)
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    k8s.v1.cni.cncf.io/networks: '[ { "name": "macvlan-conf", "ips": [ "1.1.1.247/25"
      ], "mac": "00:50:56:23:6e:bb", "gateway": [ "1.1.1.129" ] }]'
  creationTimestamp: "2022-06-10T17:26:07Z"
  name: samplepod
  namespace: playground
  resourceVersion: "906924816"
  uid: cd06da57-84e4-4422-8a16-4e9372dc0d94
spec:
  containers:
  - command:
    - /bin/bash
    - -c
    - 'trap : TERM INT; sleep infinity & wait'
    image: dougbtv/centos-network
    imagePullPolicy: Always
    name: samplepod
    ports:
    - containerPort: 80
      protocol: TCP
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-rsnnv
      readOnly: true
  dnsPolicy: ClusterFirst
  nodeName: node-04
  ...
  ```
- Other log outputs (if you use multus logging)

**Comment by s1061123** [comment]
I'll close this issue because it is not multus issue as I described. Please let me know if you find another information which may related to multus.

**Comment by mr-yaky** [comment]
Thanks for the answer. Got it.
I want to leave some details here maybe it will help someone in the future.
After checking many times with different configurations finally I found that it seems like a limitation or a restriction from docker side. Also configuring promiscuous mode for public iface didn't solve the problem and neither changing VM iface from virtio to e1000 model. Then we tried to use the same configuration on k8s nodes with CRI-O engine and it works without any problems.
Also we found that with docker and `passthru` mode for macvlan the pod with public IP was reachable but the public IP of k8s node becomes always unavailable. We didn't check with `passthru` mode for macvlan and CRI-O nodes because with `bridge` mode everything is ok.
